qcodes/dataset/sqlite/queries.py

```diff
@@ -1161,6 +1162,21 @@ def _update_run_description(conn: ConnectionPlus, run_id: int,
         conn.cursor().execute(sql, (description, run_id))
 
 
+def update_parent_datasets(conn: ConnectionPlus,
+                           run_id: int, links_str: str) -> None:
+    """
+    Update (i.e. overwrite) the parent_datasets field for the given run_id
+    """
+
+    sql = """
+          UPDATE runs
+          SET parent_datasets = ?
+          WHERE run_id = ?
+          """
+    with atomic(conn) as conn:
+        conn.cursor().execute(sql, (links_str, run_id))
```

Comment:
what is a difference between datasets that have parent_datasets equal to `NULL` and `"[]"`? if there's no difference, then shall this column be populated with `"[]"` and hence `if maybe_link_str is None: link_str = json.dumps([])` in `get_parent_dataset_links` is not needed?

Reply:
I think you are right, there should not be two different values (`NULL`/`None` and `"[]"`/`[]`) that represent the same thing. I'll make the upgrader populate the column accordingly and apply the simplification you propose.